### DISCUSSION TITLE
Batch Analysis using SLEAP CLI + GUI

### DISCUSSION THREAD

**Discussion by HaixinLiuNeuro** [talmolab/sleap#754] [Ideas]
Dear SLEAP community,

I'm new to SLEAP and just found it is exceptionally good especially for mono color movies (whisper: better result with the same labelled data comparing to DLC!) 

I'm trying to use the trained model to run batch analysis on HPC. 
I read the 'Training and inference on an example dataset' notebook, where it calls a single video file, like .mp4.
I was wondering if I have a folder of .mp4 files, is it possible to ask SLEAP to run prediction on all the video files in that folder? 
I read the function usage of sleap-track, I did not find a good answer (I might missed it) 

Thanks!

**Comment by talmo** [comment]
Hi @HaixinLiuNeuro,

Glad you've been liking SLEAP!

The simplest way to do that would be to just call `sleap-track` with the name of each video file in the folder.

Here's a quick bash script to do that on every .mp4 file without typing out the filenames individually:

```bash
for video in path/to/videos/*.mp4
do
  sleap-track -m "my/model" "$video"
done
```
Just edit the command inside the loop if you have any additional parameters you need to add.

Let me know if that works for you.

Cheers,

Talmo

**Comment by HaixinLiuNeuro** [comment]
Awesome! Thank you!

**Comment by jjbake** [comment]
Hello there,

I am a beginning programmer and I am having trouble getting this bash script to work. 

Here is how I modified the code that you wrote and I keep getting the error "video unexpected at this time"

for video in "E:\jacobE\psilocybin-HTR\output\*.mp4"do sleap-track -m "E:\jacobE\psilocybin-HTR\predictions\models\second-run220425_131610.single_instance.n=142" "$video" done

I have to run inference on about 50 videos and it would save me a ton of time if I could run inference on all of them continously. Also, this is for tracking a single animal if that changes anything. Thank you for the help.

**Comment by roomrys** [comment]
Hi @jjbake,

Assuming that the videos are stored in the `output` directory, I think you forgot the last slash in the `path/to/videos/`, try adding a slash between `output` and `*` as so:

`E:\jacobE\psilocybin-HTR\output\*.mp4`

Let us know if that works - if not, another thing to try is to either change the direction of the slashes (use forward slashes instead of backslashes as bash treats [backslashes as escape characters](https://stackoverflow.com/questions/40396613/forward-slash-vs-backward-slash-for-file-path-in-git-bash#:~:text=Bash%20treats%20backslash%20as%20an,the%20backslash%20itself%20is%20dropped.)) or use single quotes instead of double quotes around your paths.

Thanks,
Liezl

**Comment by jjbake** [comment]
I feel so silly. Fixing the dash on my directory worked. Thank you for your timely input. :)

**Comment by roomrys** [comment]
We should add documentation for this on the website.

**Comment by ARTUR551** [comment]
Hi,

Is it possible to add the feature of Batch analysis to the GUI?
That could save a lot of time to beginners like me.
And also the exporting of multi-video HDF5 files with the name of the video would be very useful, like in #729.

Thanks for developing SLEAP!

A.

**Comment by talmo** [comment]
Closing this one, but follow #733 for progress on this issue.

**Comment by ARTUR551** [comment]
Hi,

Sorry but I've been trying the solutions using the CLI (#497 and #668) and I'm not able to get working any of them.
Is it possible to get more detailed instructions for batch inference with the CLI? I'm not used to the CLI at all...

Thanks!

A.

**Comment by roomrys** [comment]
Hi @ARTUR551,

We recently added a feature that allows inference on all videos in the project #733, #749. This feature is only on the develop branch for now, so you would need to [install SLEAP from the source](https://sleap.ai/installation.html#conda-from-source). 

The option to predict on "all videos" is what you are looking for.

![image](https://user-images.githubusercontent.com/38435167/169425795-40ec165c-ee2f-4cdd-9887-e1533cac1135.png)

If you are still having trouble and/or still want to run the batch inference through the command line, I'll need a little bit more info before I can help. Do you mind providing the script you are trying to run and a description of the directory tree? Did you get any errors in the command line? 

Don't hesitate to ask! We are happy to help.

Thanks,
Liezl

**Reply by ARTUR551** [reply]
Hi Liezl,

I finally managed to predict in all videos using SLEAP from the source. I was having the error described above "video unexpected at this time" using the script, I checked carefully the path and nothing was wrong (apparently).

I finally was able to install SLEAP from the source by installing "Git", otherwise there was an error telling me that "git" was not recognized. Beginners like me could experience the same, unfortunately the GUI is very important to us, thank you for improving it constantly! I stopped using DLC because of the GUI...

Thanks!

Artur
